resources/assets/coffee/_classes/store-supporter-tag.coffee

```diff
@@ -106,42 +105,44 @@ class @StoreSupporterTag
       @searching = false
       @updateSearchResult()
 
+
   calculate: (position) =>
     new StoreSupporterTagPrice(Math.floor(position / @RESOLUTION))
 
+
   onSliderValueChanged: (event, ui) =>
     @slider.dataset.lastValue = ui.value
     @cost = @calculate(ui.value)
     @updateCostDisplay()
 
+
   onInput: (event) =>
     if !@searching
       @searching = true
       @updateSearchResult()
     @debouncedGetUser(event.currentTarget.value)
 
+
   setUserInteraction: (enabled) =>
     StoreCart.setEnabled(enabled)
     # TODO: need to elevate this element when switching over to new store design.
     $(@el).toggleClass('js-store--disabled', !enabled)
     $('.js-slider').slider('disabled': !enabled)
 
+
   sliderValue: (price) ->
     price * @RESOLUTION
 
+
   updateSearchResult: =>
     if @searching
-      @usercard.innerHTML = $('#js-usercard__loading-template').html()
+      $.publish 'store-supporter-tag:update-user', null
```

Comment:
Can't `@user` be set to `null` when setting `@searching` to `false`? Then the path would be the same between searching and not. Also the `setUserInteraction` won't need any parameter as it can just refer to `@user?.id?` directly.

Or maybe even

```coffee
setUser: (user) =>
  @user = user
  $.publish ...
  ....value = @user?.id
  setUserInteraction()
```